miio/curtain_youpin.py

```diff
@@ -0,0 +1,290 @@
+
+import click
+
+from typing import Any, Dict
+from .click_common import EnumType, command, format_output
+from .exceptions import DeviceException
+from .miot_device import MiotDevice
+
+_LOGGER = logging.getLogger(__name__)
+_MAPPING = {
+        # # Source http://miot-spec.org/miot-spec-v2/instance?type=urn:miot-spec-v2:device:curtain:0000A00C:lumi-hagl05:1
+        # Curtain
+        "motor_control": {"siid": 2, "piid": 2}, # 0 - Pause, 1 - Open, 2 - Close, 3 - auto
+        "current_position": {"siid": 2, "piid": 3}, # Range: [0, 100, 1]
+        "status": {"siid": 2, "piid": 6}, # 0 - Stopped, 1 - Opening, 2 - Closing
+        "target_position": {"siid": 2, "piid": 7}, # Range: [0, 100, 1]
+        # curtain_cfg
+        "manual_enabled": {"siid": 4, "piid": 1}, #
+        "polarity": {"siid": 4, "piid": 2},
+        "position_limit": {"siid": 4, "piid": 3},
+        "night_tip_light": {"siid": 4, "piid": 4},
+        "run_time": {"siid": 4, "piid": 5}, # Range: [0, 255, 1]
+        # motor_controller
+        "adjust_value": {"siid": 5, "piid": 1}, # Range: [-100, 100, 1]
+}
+
+# Model: ZNCLDJ21LM (also known as "Xiaomiyoupin Curtain Controller (Wi-Fi)"
+MODEL_CURTAIN_HAGL05 = "lumi.curtain.hagl05"
+
+class CurtainMiotException(DeviceException):
+    pass
+
+class MotorControl(enum.Enum):
+    Pause = 0
+    Open = 1
+    Close = 2
+    Auto = 3
+
+class Status(enum.Enum):
+    Stopped = 0
+    Opening = 1
+    Closing = 2
+
+class Polarity(enum.Enum):
+    Positive = 0
+    Reverse = 1
+
+class ManualEnabled(enum.Enum):
+    Disable = 0
+    Enable = 1
+
+class PosLimit(enum.Enum):
+    Unlimit = 0
+    Limit = 1
+
+class NightTipLight(enum.Enum):
+    Disable = 0
+    Enable = 1
+
+
+class CurtainStatus:
+    def __init__(self, data: Dict[str, Any]) -> None:
+        """ Response from device
+            {'id': 1, 'result': [
+                {'did': 'motor_control', 'siid': 2, 'piid': 2, 'code': -4001},
+                {'did': 'current_position', 'siid': 2, 'piid': 3, 'code': 0, 'value': 0},
+                {'did': 'status', 'siid': 2, 'piid': 6, 'code': 0, 'value': 0},
+                {'did': 'target_position', 'siid': 2, 'piid': 7, 'code': 0, 'value': 0},
+                {'did': 'manual_enabled', 'siid': 4, 'piid': 1, 'code': 0, 'value': 1},
+                {'did': 'polarity', 'siid': 4, 'piid': 2, 'code': 0, 'value': 0},
+                {'did': 'position_limit', 'siid': 4, 'piid': 3, 'code': 0, 'value': 0},
+                {'did': 'night_tip_light', 'siid': 4, 'piid': 4, 'code': 0, 'value': 1},
+                {'did': 'run_time', 'siid': 4, 'piid': 5, 'code': 0, 'value': 0},
+                {'did': 'adjust_value', 'siid': 5, 'piid': 1, 'code': -4000}
+            ]}
+        """
+        self.data = data
+
+    @property
+    def status(self) -> Status:
+        """ Device status """
+        return Status(self.data["status"])
+
+    """ curtain_cfg """
+    @property
+    def manual_enabled(self) -> ManualEnabled:
```

Comment:
```suggestion
    def manual_enabled(self) -> bool:
```
This would be cleaner as a simple boolean.